File: count_number_occurance.py
# List of contents
contents = [1, 2, 1, 1, 2, 3, 4]

# Initialize an empty dictionary to store counts
result_map = {}

# Iterate over the list and count occurrences of each element
for item in contents:
    if item in result_map:
        result_map[item] += 1
    else:
        result_map[item] = 1

print("M equal to:", result_map)

File: js solutions/countNumberOccurance.js
// Step 1: Initialize the Array
let contents = [1, 2, 1, 1, 2, 3, 4];

// Step 2: Initialize a Counter Object
let countObject = {};

// Step 3: Iterate through the Array
for (let num of contents) {
  // Step 4: Update Counts
  if (countObject[num]) {
    countObject[num]++;
  } else {
    countObject[num] = 1;
  }
}

// Step 5: Return the Count Object
console.log(countObject);
